peps/pep-0738.rst

```diff
@@ -0,0 +1,351 @@
+This would support `98% of active devices
+<https://dl.google.com/android/studio/metadata/distributions.json>`__, and would
+allow us to rely on a number of `dynamic linker improvements
+<https://android.googlesource.com/platform/bionic/+/refs/heads/master/android-changes-for-ndk-developers.md>`__
+which simplify the use of dynamic libraries.
+
+
+Architectures
+-------------
+
+Android currently supports 4 architectures. Their names as used by the Android
+tools are:
+
+* ``armeabi-v7a``
+* ``arm64-v8a``
+* ``x86``
+* ``x86_64``
+
+Virtually all current physical devices use one of the ARM architectures. ``x86``
+and ``x86_64`` are supported for use in emulators on development machines.
+
+``arm64-v8a`` is also supported in emulators running on macOS with Apple
+silicon. Unlike on iOS, an ARM64 emulator uses the same ABI as a real ARM64
+device, and can run the same binaries.
+
+We propose that Tier 3 support will only cover the 64-bit platforms
+(``arm64-v8a`` and ``x86_64``):
+
+* ``x86`` has not been supported as a development platform since 2020, and no
+  new emulator images have been released since then.
+* ``armeabi-v7a`` is still used, but its proportion of active devices is now
+  `less than 10% and steadily falling
+  <https://github.com/chaquo/chaquopy/issues/709#issuecomment-1744541892>`__.
+
+However, no changes should be made which would impede any downstream projects
+which still wish to build for the 32-bit architectures.
+
+
+App lifecycle
+-------------
+
+The primary programming language on Android is Java, or its modern descendant
+Kotlin. As such, an app does not provide its own executable file. Instead, all
+apps start off as a Java virtual machine running an executable provided by the
+operating system. The app's Java code can then add native code to the process by
+loading dynamic libraries and calling them through JNI.
+
+Unlike iOS, creating subprocesses *is* supported on Android. However apps may
+only run executables in `certain locations
+<https://issuetracker.google.com/issues/128554619#comment4>`__, none of which
+are writable at runtime. Long-running subprocesses are `officially discouraged
+<https://issuetracker.google.com/issues/128554619#comment4>`__, and are not
+guaranteed to be supported in future Android versions.
+
+Android does provide a command-line shell, but this is intended only for use by
+developers, and is not available to the typical end user.
+
+For these reasons, the primary way of running Python on Android will be by
+loading ``libpython3.x.so`` into the main app process. Although there will also
+be a ``python3.x`` executable linked against ``libpython3.x.so``, this is only
+for debugging, not production use.
+
+
+Specification
+=============
+
+Scope of work
+-------------
+
+The focus of this work will be to produce an Android equivalent to the existing
+Windows embeddable package. This will be used by app developers, not end users
+directly, so no installer will be required.
```

Comment:
Like the Windows embeddable package, it would simply be a ZIP file containing the compiled libraries – that gives the largest possible flexibility to downstream users. And like the Windows embeddable package, it couldn't simply be "dropped into a project" without additional work to link it with the rest of the app. I'll update the document to clarify this.